packages/swap/src/utils/statechain.ts

```diff
@@ -12,11 +13,67 @@ export type SwapRateArgs = {
   destAsset: InternalAsset;
   amount: bigint;
   limitOrders?: LimitOrders;
+  brokerCommissionBps?: number;
 };
 
 export type LimitOrders = RpcParams['cf_swap_rate_v2'][3];
 
-export const getSwapRateV2 = async ({ srcAsset, destAsset, amount, limitOrders }: SwapRateArgs) => {
+const getDeductedBrokerFeeOutput = ({
+  destAsset,
+  inputAmount,
+  intermediateAmount,
+  outputAmount,
+  brokerCommissionBps,
+  egressFee,
+}: {
+  destAsset: InternalAsset;
+  inputAmount: bigint;
+  intermediateAmount: bigint | null;
+  outputAmount: bigint;
+  brokerCommissionBps?: number;
+  egressFee: bigint;
+}) => {
+  if (!brokerCommissionBps) {
+    return { intermediateAmount, outputAmount };
+  }
+  let usdcAmount = intermediateAmount ?? inputAmount;
+  if (destAsset === 'Usdc') {
+    usdcAmount = outputAmount;
+  }
+
+  const brokerFee = BigInt(
+    brokerCommissionBps && getPipAmountFromAmount(usdcAmount, brokerCommissionBps),
+  );
+
+  const outputAmountExchangeRate = (outputAmount + egressFee) / (intermediateAmount ?? inputAmount);
+
+  if (intermediateAmount) {
+    return {
+      intermediateAmount: intermediateAmount - brokerFee,
+      outputAmount: outputAmount - brokerFee * outputAmountExchangeRate,
```

Comment:
not 100% sure if i understand the `outputAmountExchangeRate` - think we could also do something like this
```suggestion
      outputAmount: outputAmount - getPipAmountFromAmount(outputAmount, brokerCommissionBps),
```

Reply:
thanks for the suggestion, simplified the logic based on that :)